runtime/common/xwalk_switches.cc

```diff
@@ -32,6 +32,9 @@ const char kXWalkDataPath[] = "data-path";
 #if defined(OS_ANDROID)
 // Specifies the separated folder to save user data on Android.
 const char kXWalkProfileName[] = "profile-name";
+// Specifies the user data directory, which is where the browser will look for
+// all of its state.
+const char kUserDataDir[] = "user-data-dir";
```

Comment:
Yes, it is general. In this patch, the implementation is only for android, so I wrap it with OS_ANDROID. I will adjust it.